### ISSUE TITLE
Add a `check` subcommand to verify a container prepared from any image against a given spec

### ISSUE THREAD

**Issue by ok-ryoko** [ok-ryoko/turret#8]
### Code of conduct

- [X] I have read and agree to the community’s code of conduct.

### Use case(s)

There are no tests, yet this is a complex program. How are prospective users to build confidence and trust in Turret’s correct operation? How are Turret developers to ensure that their changes don’t cause regressions?

### Ideal solution

I propose the implementation of a `check` subcommand. `check` validates a container prepared from a given image against a given spec. `check` would enable Turret developers to write simple end-to-end tests for a limited set of base images, e.g.,

```sh
turret build -v 3 f38.toml
turret check localhost/turret/f38:0.1.0 f38.toml
```

`check` looks and feels like modern and more familiar test utilities. (We can take inspiration from `cargo test`, `go test`, `jest`, `pytest`, `zig test` and so on.)

A spec can contain multiple errors distributed across independent tables. Thus, I can set the `--errexit` (`-e`) option when running `check`. Like its POSIX shell namesake, `-e` tells `check` to exit on the first error. By default, `check` reports all the errors that it can find.

I can use `check` to verify that Turret can build an image to spec for any base image of my choice, even those not supported officially by Turret developers. This is key since the selection of base images is virtually unlimited.

I can use `check` to establish a series of images that satisfy the same spec. For example, suppose that:

- `a:latest`, `b:latest` and `c:latest` are images prepared from `a.toml`, `b.toml` and `c.toml`, respectively;
- `c:latest` is based on `b:latest`, and
- `b:latest` is based on `a:latest`.

Suppose further that I require all descendents of `a:latest` to satisfy `a.toml`. I can prove this by running

```sh
turret check localhost/b:latest a.toml
turret check localhost/c:latest a.toml
```

If I encounter an error, then I know I’m violating a constraint of `a.toml` somewhere in `b.toml` or `c.toml`. Conversely, I can require that all ancestors of `c:latest` satisfy `c.toml`:

```sh
turret check localhost/b:latest c.toml
turret check localhost/a:latest c.toml
```

In other words, `check` gives us a way to establish forward and backward compatibility for specs.

### Supporting information

Turret builds over a complex stack: the Linux kernel, GNU core utilities, Buildah and so on. To minimize the complexity of Turret itself, we can assume the correctness of each underlying layer and concentrate testing on system boundaries. Within this paradigm, `check` is a natural initial solution to the testing problem. However, we still need to explore Turret’s compatibility with various execution contexts—a problem for another issue.

`check` could depend on a `checker` package much like `build` already depends on the `builder` package.

`check` can’t prove that a container was created by Turret from the given spec.

**Comment by ok-ryoko** [comment]
It would be more ergonomic to invert the order of the operands. Instead of running

```console
$ turret check REFERENCE SPEC
```

… we run

```console
$ turret check SPEC REFERENCE
```

This is more consistent with the usage of the `build` subcommand, where `SPEC` follows the command name and options. Moreover, it makes it easier to (eventually) validate multiple images against the same spec:

```console
$ turret check SPEC REFERENCE...
```
